### PR TITLE
Add basic dev container setup

### PR DESCRIPTION
**Pull request by EricPedley**
Dev containers are a way to isolate all the dependencies of our project from whatever people have installed on their system. This means we can have everyone running the same versions of Ubuntu, Python, ROS, etc. when developing. If you want to set this up yourself, this PR adds instructions to the README. LMK if that environment is missing anything. I know I haven't figured out the Python installation yet but we can save that for another PR

### CODE REVIEW

**Comment by EricPedley** [comment]
@Thomas-Neill since Derek hasn't responded in a while can you verify this works on more machines than just my own and see if anything's missing?

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        "ROS_DOMAIN_ID": "42"
+    },
+    "runArgs": [
+        "--net=host",
+        "-e", "DISPLAY=${env:DISPLAY}"
+    ],
+    "forwardPorts": [
+        6080
+    ],
+    "postCreateCommand": "echo 'source /opt/ros/foxy/setup.sh' > /root/.bashrc"
```
Comment (Thomas-Neill) [inline .devcontainer/devcontainer.json:28]: Totally ignorant here but should this live in the Dockerfile?

**Review by Thomas-Neill — APPROVED** [review]
LGTM

**Comment by Thomas-Neill** [comment]
Oh I think we need to set this up so that it mounts our messages repo too. (I'm not actually sure why that is a separate repository..) Also the image build takes a long time so it would be nice to upload this to docker hub or something.

File: README.md [hunk 2]
```diff
@@ -1,172 +1,189 @@
...
+
+1. Open this project in vscode
+2. Install the "Dev Containers" extension
+3. Open the command pallete (ctrl-shift-p), then search for and execute "Dev Containers: (Re-)build and Reopen in Container"
+4. Congratulations, you get to skip all those tedious steps to install ROS 2 manually, and your environment is isolated from the rest of your computer
+5. To make downloading dependencies reproducible, add any important software installation steps to the Dockerfile in this repo.
+6. To use git inside the docker container, you may have to manually log in to GitHub again if the built-in credential forwarding isn't working. I recommend using the [GitHub CLI](https://cli.github.com/) to do this.
+7. If you want to use the simulator:
+	1. If you want to run in headless, `cd /home/ws/PX4-Autopilot` then `HEADLESS=1 make px4_sitl gazebo-classic`
+	2. If you want it to run it in a GUI, one way is using the remote desktop environment in the dev container. Open `localhost:6080` in a web browser, then enter password `vscode`, then use the menu in the bottom left to open a terminal, `cd /home/ws/PX4-Autopilot`, then run `make px4_stil gazebo-classic`.
```
Comment (Thomas-Neill) [inline README.md:41]:
```suggestion
	2. If you want it to run it in a GUI, one way is using the remote desktop environment in the dev container. Open `localhost:6080` in a web browser, then enter password `vscode`, then use the menu in the bottom left to open a terminal, `cd /home/ws/PX4-Autopilot`, then run `make px4_sitl gazebo-classic`.

```

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (EricPedley) [reply .devcontainer/devcontainer.json]: no clue either, but it's probably not a big deal and easy to move later.

**Comment by EricPedley** [comment]
@Thomas-Neill @Herpderk Do we need to be using gazebo classic or can we upgrade to garden? I tried building this shit with ROS humble like we discussed in discord and now the SITL is breaking. Also most of the docs I've read online recommend Garden.

**Comment by EricPedley** [comment]
Just putting this back to foxy for now, and we can deal with humble migration later

**Pull request merged**